--- axiom/store.py.orig	2009-05-02 11:06:28.000000000 +0100
+++ axiom/store.py	2009-05-02 11:13:51.000000000 +0100
@@ -1411,7 +1411,7 @@
         # from the schema and the sorted() here doesn't seem to be any faster
         # than doing this.
         persistedSchema = sorted(self.querySchemaSQL(
-            "SELECT attribute, type_id, sqltype, indexed, "
+            'SELECT attribute, type_id, sqltype, "indexed", '
             "pythontype, docstring FROM *DATABASE*.axiom_attributes "))
 
         # This is trivially (but measurably!) faster than getattr(attributes,
